### PR TITLE
electrum: 3.2.4 -> 3.3.2 plus new dependencies

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

Previous PR was closed, I applied required fixes
:warning:  I did not test execution at this time

Also the aio libraries are missing tests, maybe they are not included in the Pypi package

cc @clefru @dotlambda 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by clefru** [comment]
> Previous PR was closed, I applied required fixes

LGTM. Thank you!

>  I did not test execution at this time

Ran result/bin/electrum and created a test wallet. Works for me.

**Comment by clefru** [comment]
@GrahamcOfBorg build electrum

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5193,6 +5195,8 @@ in {
 
   casttube = callPackage ../development/python-modules/casttube { };
 
+  aiorpcx = callPackage ../development/python-modules/aiorpcx { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: This has to be moved up.

**Review by dotlambda — CHANGES_REQUESTED** [review]
> Also the aio libraries are missing tests, maybe they are not included in the Pypi package

Yes, the tests/ directory is missing in both tarballs. Please add `doCheck = false` and a comment with the reason.

**Comment by dotlambda** [comment]
Also don't forget to squash your commits such that only three are remaining.

**Comment by nyanloutre** [comment]
@dotlambda I could also fetch the libraries from GitHub to get the tests

**Comment by dotlambda** [comment]
> @dotlambda I could also fetch the libraries from GitHub to get the tests

That's up to you. However, I suspect at least the aiohttp-socks tests to fail due to accessing the network.

**Comment by nyanloutre** [comment]
@dotlambda even worse the tests depend on a binary contained in the repo

**Comment by nyanloutre** [comment]
@dotlambda I enabled the tests for electrum but I am not sure if it will work on MacOS. Could you trigger a borg build ?

**Comment by nyanloutre** [comment]
I will squash commits once everything is validated

**Comment by veprbl** [comment]
@GrahamcOfBorg build electrum

File: pkgs/applications/misc/electrum/default.nix [hunk 2]
```diff
@@ -65,7 +70,11 @@ python3Packages.buildPythonApplication rec {
...
   '';
 
-  doCheck = false;
+  checkInputs = with python3Packages; [ pytest ];
+
+  checkPhase = ''
+    py.test electrum/tests
+  '';
 
   doInstallCheck = true;
```
Comment (dotlambda) [inline pkgs/applications/misc/electrum/default.nix]: Please merge `installCheckPhase` into `checkPhase`.

File: pkgs/development/python-modules/aiohttp-socks/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "aiohttp_socks";
+    sha256 = "0473702jk66xrgpm28wbdgpnak4v0dh2qmdjw7ky7hf3lwwqkggf";
+  };
+
+  propagatedBuildInputs = [ aiohttp ];
+
+  # Checks needs internet access
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiohttp-socks/default.nix]: I don't think so, but they use localhost, to which access is denied in the sandbox.

File: pkgs/development/python-modules/aiorpcx/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    inherit version;
+    pname = "aiorpcX";
+    sha256 = "1p88k15jh0d2a18pnnbfcamsqi2bxvmmhpizmdlxfdxf8vy5ggyj";
+  };
+
+  propagatedBuildInputs = [ attrs ];
+
+  disabled = pythonOlder "3.6";
+
+  # Checks needs internet access
```
Comment (dotlambda) [inline pkgs/development/python-modules/aiorpcx/default.nix]: no

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/aiohttp-socks/default.nix (hunk 3 above)
Comment (nyanloutre) [reply pkgs/development/python-modules/aiohttp-socks/default.nix]: Isn't this a remote address ? https://github.com/romis2012/aiohttp-socks/blob/3252f4bdd37fb9a7360481977f800189cb3e3aca/tests/test_socks.py#L18

File: pkgs/development/python-modules/aiorpcx/default.nix (hunk 4 above)
Comment (nyanloutre) [reply pkgs/development/python-modules/aiorpcx/default.nix]: I spotted a remote address used here : https://github.com/kyuupichan/aiorpcX/blob/d8568f6c1865d1927034e05807a412b08c6d7526/tests/test_socks.py#L15

File: pkgs/development/python-modules/aiohttp-socks/default.nix (hunk 3 above)
Comment (dotlambda) [reply pkgs/development/python-modules/aiohttp-socks/default.nix]: Indeed, it is. Sorry.

**Review by dotlambda — APPROVED** [review]
LGTM, apart from squashing the commits. Did not test any functionality.

**Comment by nyanloutre** [comment]
@dotlambda do you want it to be more squashed ?

**Comment by dotlambda** [comment]
The last two could be squashed together :)

**Comment by nyanloutre** [comment]
done

**Pull request merged**